### PR TITLE
refactor(capnp, thrift): improve grammar & queries to stable for the foreseeable future

### PR DESCRIPTION
**Pull request by amaanq**
Apologies for the rapid flurry of PR's, I hadn't been able to get local highlights to reflect their change on demand after a parser/query update. Finally figured that out today and I've gotten a lot more improvements done as a result.

Thanks!
-Amaan

### CODE REVIEW

File: queries/capnp/highlights.scm [hunk 1]
```diff
@@ -82,11 +88,14 @@
   "extends"
   "namespace"
   "using"
+  (void)
```
Comment (theHamsta) [inline queries/capnp/highlights.scm:91]: `@type.builtin` or `@keyword`? I don't know `void` in the context of capnp

File: queries/capnp/highlights.scm [hunk 2]
```diff
@@ -119,6 +129,8 @@
   "->"
 ] @punctuation.delimiter
 
+(data_hex) @symbol
```
Comment (theHamsta) [inline queries/capnp/highlights.scm:132]: data doesn't sound like `@symbol`. `@symbol` is rather something like a named constant in the field of functional languages

File: queries/thrift/highlights.scm [hunk 3]
```diff
@@ -47,22 +47,24 @@
...
 (exception_identifier) @type
 (exception_param_type) @type
 (field_type) @type
+(param_type) @type
 (type_identifier) @type
 
 ; Attributes
 
 (annotation_identifier) @attribute
+(uri_def) @attribute
```
Comment (theHamsta) [inline queries/thrift/highlights.scm:60]: is uri here like an annotation to something?

File: queries/capnp/highlights.scm (hunk 2 above)
Comment (amaanq) [reply queries/capnp/highlights.scm:133]: Capnp datas are unique, e.g.
```capnp
  dataField      @13 : Data    = 0x"62 61 72"; # "bar"
```

this is how you would write raw binary types, like b"" in python, so this *specific* leading 0x imo should be a special kind of highlight. what do you think?

File: queries/thrift/highlights.scm (hunk 3 above)
Comment (amaanq) [reply queries/thrift/highlights.scm:60]: it's used in special namespace defs, like so
```thrift
namespace xsd test (uri = 'http://thrift.apache.org/ns/OneWayTest')
```

File: queries/capnp/highlights.scm (hunk 1 above)
Comment (amaanq) [reply queries/capnp/highlights.scm]: it's actually @constant.builtin, just fixed it in latest commit!
Void is the type, void the value :)

File: queries/capnp/highlights.scm (hunk 2 above)
Comment (theHamsta) [reply queries/capnp/highlights.scm:133]: @number ?

Comment (amaanq) [reply queries/capnp/highlights.scm:133]: hmm, @number *just* for the 0x prefix is weird..but if thats convention sure, let me know

Comment (amaanq) [reply queries/capnp/highlights.scm:133]: @string.special works for me, let me update

Comment (amaanq) [reply queries/capnp/highlights.scm:133]: nevermind, i have that defined for the inner data string, it'd be weird to have them both be that

**Pull request merged**